### DISCUSSION TITLE
On tfcmt possible arguments and how the arguments are processed

### DISCUSSION THREAD

**Discussion by yener-azs** [suzuki-shunsuke/tfcmt#1033] [Deprecated (Question and Answer)] [answered]
### Question

Hello. I found that in our project `tfcmt` is used in our IaC pipelines. I would have a question on possible arguments that the `tfcmt` can receive. I checked this page https://suzuki-shunsuke.github.io/tfcmt/usage . So the syntax of the `tfcmt` command is `tfcmt [global options] command [command options] [arguments...]`. I would have these questions: 
1) In this example, https://suzuki-shunsuke.github.io/tfcmt/usage#tfcmt-plan , it is mentioned `tfcmt plan - Run terraform plan and post a comment to GitHub commit or pull request`. What does `$ tfcmt plan -- terraform plan` do then?
2) So in the previous example `terraform plan` is an argument (it is positioned after `--`). What other are arguments are possible for `tfcmt`?
3) Maybe because questions 1 and 2 are not clear to me, is it possible to have`$ tfcmt plan -- terraform apply`? Or `$ tfcmt apply -- terraform plan` ?
Thank you.

### Example Code

```console
$ 
```

Configuration

```yaml

```

### Reference

*

**Comment by suzuki-shunsuke** [comment, marked as answer]
tfcmt executes a command with arguments after `--`.

So `tfcmt plan -- terraform plan` executes a command `terraform plan`.
You can use any terraform options freely.

e.g.

```sh
tfcmt plan -- terraform plan -detailed-exitcode -parallelism=20
```

You can also execute a command other than `terraform plan`.

e.g.

```sh
terraform plan > result.txt 2>&1
tfcmt plan -- cat result.txt
```

> Maybe because questions 1 and 2 are not clear to me, is it possible to have$ tfcmt plan -- terraform apply? Or $ tfcmt apply -- terraform plan ?

We don't assume the use case at all.
So I'm not sure if tfcmt works well.
Maybe tfcmt can't parse the output of the command well.

Could you explain why you want to run `tfcmt plan -- terraform apply` and `tfcmt apply -- terraform plan`.
I don't think this is a common use case.
